.github/workflows/Tests.yaml

```diff
@@ -433,3 +433,87 @@ jobs:
           # We get occasional random timeouts, repeat tests to see if
           # it is a random timeout or systematic
           ctest -j2 --output-on-failure -L inputfiles --repeat after-timeout:3
+
+  arch_datastructures_tests:
+    name: Archs
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler:
+          - "gcc-7"
```

Comment:
(nitpicking, not requesting this change) The quotes `"` are not needed in this list (I included them in my original suggestion for the arch list because those started with a dash `-`, so I wasn't sure if yaml could handle the syntax)